### PR TITLE
progressbar: v.3 added new changes

### PR DESCRIPTION
**Pull request by raynesmike**

### CODE REVIEW

**Comment by raynesmike** [comment]
ProgressBar1

File: src/components/progressBar/ProgressBar.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import {SafeAreaView, StatusBar, Text, View} from 'react-native';
+import {styles} from './progressBar.style';
+
+export const ProgressBar = ({value = '5%', max = '100'}) => {
```
Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:5]: Please set by default 0

Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:5]: use only numbers. don't use string with percentige

Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:5]: and number doesn't need quoutes

File: src/components/progressBar/ProgressBar.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  return (
+    <View style={styles.container}>
+      <View style={[styles.frontBar, {width: `${max}%`}]} />
```
Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:8]:
```
const  _max =  max > 100  ? 100 : max

```
`const progressIndicatorStyle  = {
`width: `${_max}%`
}`

File: src/components/progressBar/ProgressBar.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+      <View style={[styles.backBar, {width: `${value}%`}]} />
```
Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:9]: porgressIndicaterContainer - would is a better way.

File: src/components/progressBar/progressBar.style.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+export const styles = {
+  container: {
+    flex: 1,
+    margin: 20,
+    justifyContent: 'center',
+  },
+  backBar: {
+    height: 15,
+    width: '5%',
```
Comment (lasharela) [inline src/components/progressBar/progressBar.style.js:9]: minWidth: "5%"

**Review by lasharela — CHANGES_REQUESTED** [review]

File: src/components/progressBar/ProgressBar.js (hunk 1 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: added the changes for v.2

File: src/components/progressBar/ProgressBar.js (hunk 3 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: added the changes for v.2

File: src/components/progressBar/ProgressBar.js (hunk 1 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: added the changes for v.2

File: src/components/progressBar/ProgressBar.js (hunk 2 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: added the changes

File: src/components/progressBar/ProgressBar.js (hunk 1 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: added the changes

**Review by Dulaya — COMMENTED** [review]
It works fine for me. I don't know exactly Lasha's criteria, but I was able to build a progress bar with more intuitive solution. Let me know if you want to see it. 

Additionally, I'm not a fan of how you calculate progressValue. The ternary operator works great when you render React components, but it just looks weird when you use it to calculate values like that. I'm more of a fan of either if/else or switch/case.  

I do have one concern. What if the user wants different size, color, length, etc? Are you still working on the custom user input? Otherwise it works great on my end.

File: src/components/progressBar/ProgressBar.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import {View} from 'react-native';
+import {styles} from './progressBar.style';
+
+export const ProgressBar = ({percentage = 0, newMax}) => {
```
Comment (mshago) [inline src/components/progressBar/ProgressBar.js:5]: Are you going to use newMax?

Comment (lasharela) [reply src/components/progressBar/ProgressBar.js:5]: remove this prop. not needed.

File: src/components/progressBar/ProgressBar.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    percentage < 0
+      ? 0
+      : percentage > 100
+      ? 100
+      : percentage * ((max - 2) / max);
+  const tempProgressIndicatorBar = [
+    styles.progressIndicatorBar,
+    {width: `${progressValue}%`},
+  ];
+  const tempEmptyBar = [styles.emptyBar];
```
Comment (lasharela) [inline src/components/progressBar/ProgressBar.js:17]: [styles.emptyBar] - asign this directly

File: src/components/progressBar/progressBar.style.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import {color} from '../../theme/style';
```
Comment (lasharela) [inline src/components/progressBar/progressBar.style.js:1]: in theme we have index. you can import everything from "../../theme" - that's the idea to have an index.

File: src/components/progressBar/progressBar.style.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import {size} from '../../theme/size';
```
Comment (lasharela) [inline src/components/progressBar/progressBar.style.js:2]: the same. import from index.

File: src/components/progressBar/progressBar.style.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+export const styles = {
+  container: {
+    margin: 20,
```
Comment (lasharela) [inline src/components/progressBar/progressBar.style.js:6]: try to use not fixed numbers but closest value from `size` object

File: src/components/progressBar/progressBar.style.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    borderTopRightRadius: size.sm,
+    borderBottomRightRadius: size.sm,
+    overflow: 'hidden',
+    position: 'absolute',
+  },
+  emptyBar: {
+    height: size.sm,
+    width: '100%',
+    backgroundColor: color.midGray,
+    borderRadius: 5,
```
Comment (lasharela) [inline src/components/progressBar/progressBar.style.js:30]: try to use `size`

**Review by lasharela — CHANGES_REQUESTED** [review]

File: src/components/progressBar/progressBar.style.js (hunk 10 above)
Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js]: thank you,

File: src/components/progressBar/ProgressBar.js (hunk 6 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: thank you,

File: src/components/progressBar/progressBar.style.js (hunk 7 above)
Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js:1]: I tried doing this

Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js:1]: import {color, size} from '../../theme'; but i keep getting error

File: src/components/progressBar/progressBar.style.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+export const styles = {
+  container: {
+    flex: 1,
```
[the remark this answers is not in the document]
Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js]: I removed it.

File: src/components/progressBar/ProgressBar.js (hunk 5 above)
Comment (raynesmike) [reply src/components/progressBar/ProgressBar.js]: I will remove it now. Thanks

File: src/components/progressBar/progressBar.style.js (hunk 7 above)
Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js]: solved the issue, it was the naming of the folder

File: src/components/progressBar/progressBar.style.js (hunk 8 above)
Comment (raynesmike) [reply src/components/progressBar/progressBar.style.js]: solved the issue, it was the naming of the folder

**Review by mshago — APPROVED** [review]

**Review by lasharela — APPROVED** [review]

**Review by Nap1400 — APPROVED** [review]

**Pull request merged**